pdc/apps/common/viewsets.py

```diff
@@ -186,11 +191,22 @@ def initial(self, request, *args, **kwargs):
         if (request.method.lower() not in self.http_method_names or
                 not hasattr(self, request.method.lower())):
             return
-
         allowed_keys = drf_introspection.get_allowed_query_params(self)
         extra_keys = set(request.query_params.keys()) - allowed_keys
         if extra_keys:
             raise FieldError('Unknown query params: %s.' % ', '.join(sorted(extra_keys)))
+        # If 'ordering' in query parameter, check the key whether in fields.
+        if 'ordering' in request.query_params.keys():
+            self._check_ordering_keys(request)
+
+    def _check_ordering_keys(self, request):
+        ordering_keys = request.query_params.get('ordering')
+        tmp_list = [param.strip() for param in ordering_keys.split(',')]
```

Comment:
Change `param.strip()` to `param.strip().lstrip('-')` here then it don't have to use lstrip in following code.